actions/dynamicdns

```diff
@@ -0,0 +1,443 @@
+# as plinth will add them
+doReplaceVars()
+{
+    local url=$(echo "${updateurl}" | sed "s/<Ip>/${wanip}/g")
+    url=$(echo "${url}" | sed "s/<Domain>/${host}/g")
+    url=$(echo "${url}" | sed "s/<User>/${user}/g")
+    url=$(echo "${url}" | sed "s/<Pass>/${pass}/g")
+    url=$(echo "${url}" | sed "s/'//g")
+    updateurl=${url}
+    logger "expanded update URL as ${url}"
+}
+
+# doReadCFG() needs to be run before this
+# this function will return all configured parameters in a way that
+# plinth will understand (plinth know the order of
+# parameters this function will return)
+doStatus()
+{
+    PROC=$(pgrep ${TOOLNAME})
+    if [ -f "${CRONJOB}" ];then
+        echo "${ENABLED_STRING}"
+    elif [ ! -z "${PROC}" ];then
+        echo "${ENABLED_STRING}"
+    else
+        echo "${DISABLED_STRING}"
+    fi
+    if [ ! -z "${server}" ];then
+        echo "${server}"
+    else
+        echo "${DISABLED_STRING}"
+    fi
+    if [ ! -z "${host}"  ];then
+        echo "${host}"
+    else
+        echo "${DISABLED_STRING}"
+    fi
+    if [ ! -z "${user}" ];then
+        echo "${user}"
+    else
+        echo "${DISABLED_STRING}"
+    fi
+    if [ ! -z "${pass}"  ];then
+        echo "${pass}"
+    else
+        echo "${DISABLED_STRING}"
+    fi
+    if [ ! -z "${ipurl}" ];then
+        echo "${ipurl}"
+    else
+        echo "${DISABLED_STRING}"
+    fi
+    if [ ! -z "${updateurl}"  ];then
+        echo "${updateurl}"
+    else
+        echo "${DISABLED_STRING}"
+    fi
+    if [ ! -z "${basicauth}" ];then
+        echo "${basicauth}"
+    else
+        echo "${DISABLED_STRING}"
+    fi
+    if [ ! -z "${ignoreCertError}" ];then
+        echo "${ignoreCertError}"
+    else
+        echo "${DISABLED_STRING}"
+    fi
+}
+
+# ask a public WEB Server for the WAN IP we are comming from
+# and store this ip within $wanip
+doGetWANIP()
+{
+    if [ ! -z "${ipurl}" ];then
+        outfile=$(mktemp)
+        local cmd="${WGET} ${WGETOPTIONS} -O ${outfile} ${ipurl}"
+        $cmd
+        wanip=$(sed s/[^0-9.]//g "${outfile}")
+        rm "${outfile}"
+        [ -z "${wanip}" ] && wanip=${NOIP}
+    else
+        # no WAN IP found because of missing check URL
+        wanip=${NOIP}
+    fi
+}
+
+# actualy do the update (using wget or ez-ipupdate or even both)
+# this function is called via cronjob
+doUpdate()
+{
+    local dnsentry=$(nslookup "${host}"|tail -n2|grep A|sed s/[^0-9.]//g)
+    if [ "${dnsentry}" =  "${wanip}" ];then
+        return
+    fi
+    if [ ! -z "${server}" ];then
+        start-stop-daemon -S -x "${UPDATE_TOOL}" -m -p "${PIDFILE}" -- -c "${cfgfile}"
+    fi
+    if [ ! -z "${updateurl}" ];then
+        doReplaceVars
+        if [ "${basicauth}" = "enabled" ];then
+            local wgetoptions=" --user ${user} --password ${pass} "
+        fi
+        if [ "${ignoreCertError}" = "enabled" ];then
+            local wgetoptions=" --no-check-certificate "
```

Comment:
If wgetoptions was already defined on line 303, should the --no-check-certificate option be concatenated to the existing variable (instead of replacing it)?